### PR TITLE
Build and test on Windows yay

### PR DESCRIPTION
**Pull request by abayer**
* JENKINS issue(s):
    * n/a
* Description:
    * Thought I'd finally get the tests all able to pass on Windows, and add Windows to the Jenkinsfile run. Down the road (once ci.jenkins.io is on Declarative 1.2 or later and once the junit-per-step stuff is done), I'll switch to running Windows and Linux builds in parallel.
* Documentation changes:
    * n/a
* Users/aliases to notify:
    * @reviewbybees 
    * @jtnord because it's Windows-related so I feel an obligation to namecheck him

### CODE REVIEW

File: pipeline-model-definition/src/test/resources/envDollarQuotes.groovy [hunk 1]
```diff
@@ -39,10 +39,10 @@ pipeline {
             }
 
             steps {
-                sh 'echo "FOO is $FOO"'
-                sh 'echo "BAR is $BAR"'
-                sh 'echo "BAZ is $BAZ"'
-                sh 'echo "SPLODE is $SPLODE"'
+                echo "FOO is $FOO"
```
Comment (rsandell) [inline pipeline-model-definition/src/test/resources/envDollarQuotes.groovy]: This is not really the same. In the former version you tested variable expansion in `sh` this one tests it in groovy-cps.

Comment (rsandell) [reply pipeline-model-definition/src/test/resources/envDollarQuotes.groovy]: But that probably doesn't matter.

**Comment by rsandell** [comment]
:bee:

File: pipeline-model-definition/src/test/resources/envDollarQuotes.groovy (hunk 1 above)
Comment (abayer) [reply pipeline-model-definition/src/test/resources/envDollarQuotes.groovy]: Very good point. @jtnord or anyone else - how would I write to a file from `bat`?

Comment (rsandell) [reply pipeline-model-definition/src/test/resources/envDollarQuotes.groovy]: `echo FOO is %FOO% > somefile.txt`

Comment (abayer) [reply pipeline-model-definition/src/test/resources/envDollarQuotes.groovy]: Okiedokie.

Comment (abayer) [reply pipeline-model-definition/src/test/resources/envDollarQuotes.groovy]: Oh wait, that's for a different test. Ehhhh, I'll update 'em both.

File: Jenkinsfile [hunk 2]
```diff
@@ -25,31 +22,50 @@ pipeline {
...
             }
+            post {
+                // No matter what the build status is, run this step. There are other conditions
+                // available as well, such as "success", "failed", "unstable", and "changed".
+                always {
+                    junit '*/target/surefire-reports/*.xml'
+                }
+                success {
+                    archive "**/target/*.hpi"
+                    archive "**/target/site/jacoco/jacoco.xml"
```
Comment (jtnord) [inline Jenkinsfile]: why not use the jaccoco plugin?  Also generally better to run with and without coverage in a different lifecycle...
But this is not anything new so - Meh...

**Comment by jtnord** [comment]
Code looks ok, left some drive by comments  - but [failing tests](https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fpipeline-model-definition-plugin/detail/PR-202/12/tests) :-)

File: Jenkinsfile (hunk 2 above)
Comment (abayer) [reply Jenkinsfile]: Simple answer - at the time we added that, the jacoco plugin didn't have a `@Symbol` yet. =) But it does now, so I'll add that.

Comment (abayer) [reply Jenkinsfile]: Check that, it still hasn't been released yet.

**Pull request merged**